src/super_gradients/training/models/detection_models/pp_yolo_e/pp_yolo_e.py

```diff
@@ -14,11 +17,11 @@
 from super_gradients.training.models.detection_models.pp_yolo_e.post_prediction_callback import PPYoloEPostPredictionCallback, DetectionPostPredictionCallback
 from super_gradients.training.models.prediction_results import ImagesDetectionPrediction
 from super_gradients.training.pipelines.pipelines import DetectionPipeline
-from super_gradients.training.transforms.processing import Processing
+from super_gradients.training.processing.processing import Processing
 from super_gradients.training.utils.media.image import ImageSource
 
 
-class PPYoloE(SgModule):
+class PPYoloE(SgModule, HasPredict):
```

Comment:
I would prefer to avoid double inheritance. In this case this can be handled with protocols:

```
from typing import Protocol, runtime_checkable


@runtime_checkable
class MyInterface(Protocol):
    def method1(self) -> None:
        ...

    def method2(self) -> None:
        ...


class ClassRespectingInterface:
    def method1(self):
        pass

    def method2(self):
        pass


class ClassNOTRespectingInterface:
    def method1(self):
        pass


print("ClassRespectingInterface: ", isinstance(ClassRespectingInterface(), MyInterface))
print("ClassNOTRespectingInterface: ", isinstance(ClassNOTRespectingInterface(), MyInterface))
```
```
ClassRespectingInterface:  True
ClassNOTRespectingInterface:  False
```

Reply:
@BloodAxe WDYT?
Also after talking to @Louis-Dupont , we were wondering why you did not make the module interfaces abstract classes to force the methods implementations ?

Reply:
To start, I was not aware of typing.Protocol :)
Second, Protocol supported since python 3.8. I beleive we still want to support 3.7, so Protocol is not an option yet :/
But it's looking good and https://jellis18.github.io/post/2022-01-11-abc-vs-protocol/ explains quite nicely the difference between ABC and protocols and if we decide to ditch 3.7 support I would be quite happy to  migrate to them.
If the double inheritance is a concernt, we don't have to use (ABC) as base class. We still can use decorators @abstractmethod to mark methods as abstract and put their default implementation manually:
```
class MyInterface):
    @abstractmethod
    def method1(self) -> None:
        raise NotImplementedError()
```

Reply:
Nice catch!
Seems that typing-extensions supports Protocol and runtime_checkable this 3.7 and up though.
Any objections for using them?
I do like @Louis-Dupont suggestion, as we might want to have multiple interfaces to obey to in the future.

Reply:
Sounds good to me 👍 to use `typing-extensions`

Reply:
changed in https://github.com/Deci-AI/super-gradients/pull/835/commits/d9e3019ce07736224b9e07687bc770e793a2c612